### PR TITLE
setup nodes: handle exceptions and stop the test in case it failed

### PR DESCRIPTION
**Pull request by larisau**
added wrapper method to use the same code for all node types

### CODE REVIEW

File: sdcm/cluster_gce.py [hunk 1]
```diff
@@ -359,35 +365,16 @@ def _node_setup(self, node):
         node.remoter.run('sudo cat /etc/scylla.d/io.conf')
         node.wait_db_up()
         node.wait_jmx_up()
+        node.remoter.run('sudo yum install -y {}-gdb'.format(node.scylla_pkg()),
+                         verbose=verbose, ignore_status=True)
 
     def wait_for_init(self, node_list=None, verbose=False, timeout=None):
```
Comment (amoskong) [inline sdcm/cluster_gce.py]: why don't use @wait_for_init_wrap for this wiat_for_init()?

Comment (larisau) [reply sdcm/cluster_gce.py]: It's called here - via parent wait_for_init call(the next line)

File: sdcm/cluster_gce.py [hunk 2]
```diff
@@ -359,35 +365,16 @@ def _node_setup(self, node):
...
-
-        for node in node_list:
-            dst_nodes = [n for n in node_list if n.dc_idx != node.dc_idx]
-            local_nodes = [n for n in node_list if n.dc_idx == node.dc_idx and n != node]
-            if self._param_enabled('enable_tc'):
+        if self._param_enabled('enable_tc'):
+            for node in self.nodes:
+                dst_nodes = [n for n in self.nodes if n.dc_idx != node.dc_idx]
+                local_nodes = [n for n in self.nodes if n.dc_idx == node.dc_idx and n != node]
                 self.set_tc(node, dst_nodes, local_nodes)
```
Comment (larisau) [inline sdcm/cluster_gce.py]: @amoskong, a question about set_tc: this feature can be enabled for gce only?

Comment (amoskong) [reply sdcm/cluster_gce.py]: Yes for now.

File: sdcm/cluster_gce.py (hunk 1 above)
Comment (amoskong) [reply sdcm/cluster_gce.py]: OK

**Comment by roydahan** [comment]
@amoskong, please backport this one to branch-perf.

later after some mileage with this, we will backport also to branch-2.1

**Comment by roydahan** [comment]
@amoskong, please backport to both branches.

**Pull request merged**